lib/matplotlib/tests/test_streamplot.py

```diff
@@ -114,3 +115,53 @@ def test_streamplot_limits():
     # datalim.
     assert_array_almost_equal(ax.dataLim.bounds, (20, 30, 15, 6),
                               decimal=1)
+
+
+def test_streamplot_grid():
+    u = np.ones((2, 2))
+    v = np.zeros((2, 2))
+
+    # Test for same rows and columns
+    x = np.array([[10, 20], [10, 30]])
+    y = np.array([[10, 10], [20, 20]])
+
+    msg = "The rows of 'x' must be equal"
+    with pytest.raises(ValueError, match=msg):
```

Comment:
I'd rather inline the message in the function. Having a separate variable is an unnecessary indirection.

Reply:
Something like I just pushed?